lib/trip.rb

```diff
@@ -46,9 +56,20 @@ def inspect
         "PassengerID=#{passenger&.id.inspect}>"
     end
 
-    def connect(passenger)
+    def connect(passenger, driver)
       @passenger = passenger
+      @driver = driver
       passenger.add_trip(self)
+      driver.add_trip(self)
+    end
+
+    def duration
+      if end_time != nil
```

Comment:
Nice work handling in progress trips across the board (especially testing for this edge case!). I'm not sure if this would make sense with the rest of your implementation, but you might consider whether raising an exception instead of returning a string could work for in progress trips.